dogu/versionRegistry.go

```diff
@@ -0,0 +1,423 @@
+				}
+				throwAndLogWatchError(ctx, fmt.Errorf("watch event type is error: %q", status.String()), currentVersionsWatchResult)
+				return
+			}
+		}
+	}
+}
+
+func handleDeleteWatchEvent(event watch.Event, persistenceContext map[SimpleDoguName]core.Version, currentVersionsWatchResult chan CurrentVersionsWatchResult) error {
+	specConfigMap, err := getSpecConfigMapFromEvent(event)
+	if err != nil {
+		return err
+	}
+
+	if !hasDoguSpecConfigMapCurrentKey(specConfigMap) {
+		// disabled dogus deleted. Do nothing
+		return nil
+	}
+
+	eventDoguVersion, err := getCurrentDoguVersionFromDoguSpecConfigMap(*specConfigMap)
+	if err != nil {
+		return err
+	}
+
+	oldPersistenceContext := copyPersistenceContext(persistenceContext)
+	delete(persistenceContext, eventDoguVersion.Name)
+
+	fireWatchResult(currentVersionsWatchResult, oldPersistenceContext, persistenceContext, []DoguVersion{eventDoguVersion})
+	return nil
+}
+
+func handleModifiedWatchEvent(ctx context.Context, event watch.Event, persistenceContext map[SimpleDoguName]core.Version, currentVersionsWatchResult chan CurrentVersionsWatchResult) error {
+	logger := log.FromContext(ctx).WithName("VersionRegistry.handleModifiedWatchEvent")
+	specConfigMap, err := getSpecConfigMapFromEvent(event)
+	if err != nil {
+		return err
+	}
+
+	oldPersistenceContext := copyPersistenceContext(persistenceContext)
+
+	// Skip process. Configmap was possible created empty and will get modified event on Enable.
+	if !hasDoguSpecConfigMapCurrentKey(specConfigMap) {
+		// Dogu was disabled
+		doguName := SimpleDoguName(specConfigMap.Labels[doguNameLabelKey])
+		version, ok := oldPersistenceContext[doguName]
+		if !ok {
+			// Dogu ist still disabled and cm got other updates than current deletion
+			return nil
+		}
+		fireWatchResult(currentVersionsWatchResult, oldPersistenceContext, persistenceContext, []DoguVersion{{Name: doguName, Version: version}})
+		delete(persistenceContext, doguName)
+	} else {
+		// Detect change
+		eventDoguVersion, getErr := getCurrentDoguVersionFromDoguSpecConfigMap(*specConfigMap)
+		if getErr != nil {
+			return getErr
+		}
+
+		version, ok := persistenceContext[eventDoguVersion.Name]
+		if ok && version.IsEqualTo(eventDoguVersion.Version) {
+			logger.Info("current versions %s for dogu %s from persistent context and modified event are equal", eventDoguVersion.Version.Raw, eventDoguVersion.Name)
+			return nil
+		}
+
+		persistenceContext[eventDoguVersion.Name] = eventDoguVersion.Version
+		fireWatchResult(currentVersionsWatchResult, oldPersistenceContext, persistenceContext, []DoguVersion{eventDoguVersion})
+	}
+
+	return nil
+}
+
+func handleAddWatchEvent(event watch.Event, persistenceContext map[SimpleDoguName]core.Version, currentVersionsWatchResult chan CurrentVersionsWatchResult) error {
+	specConfigMap, err := getSpecConfigMapFromEvent(event)
+	if err != nil {
+		return err
+	}
+
+	// Skip process. Configmap was possible created empty and will get modified event on Enable.
```

Comment:
But this is an add event?

Reply:
Yes. A creation of a empty cm. Maybe the last part of the sentence is just confusing